helpers/helpers.go

```diff
@@ -396,34 +398,44 @@ func CheckTokenExpiration(accessToken string) error {
 }
 
 // ListFiles returns a list for s3 objects that correspond to files with the specified prefix.
-func ListFiles(config Config, prefix string) ([]*s3.Object, error) {
-	sess := session.Must(session.NewSession(&aws.Config{
-		// The region for the backend is always the specified one
-		// and not present in the configuration from auth - hardcoded
-		Region: aws.String("us-west-2"),
-		Credentials: credentials.NewStaticCredentials(
+func ListFiles(config Config, prefix string) ([]types.Object, error) {
+	s3Endpoint := config.HostBase
+	if !strings.HasPrefix(s3Endpoint, "http") {
+		s3Endpoint = fmt.Sprintf("http://%s", config.HostBase)
```

Comment:
```suggestion
		if config.UseHTTPS {
			s3Endpoint = fmt.Sprintf("https://%s", config.HostBase)
		} else {
			s3Endpoint = fmt.Sprintf("http://%s", config.HostBase)
		}   
```
Current code overwrite the prefix to `https://`  only when `UseHTTPS` is `false`